text/0000-es6-modules.md

```diff
@@ -0,0 +1,122 @@
+- Start Date: (2015-06-16)
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Instead of accessing functions and classes directly on the `Ember` namespace,
+move all the public API under ES6 modules.
+
+# Motivation
+
+As build tooling improved during the recent years, especially with Ember CLI, it
+is now fairly straightforward to rely only on ES6 modules. This comes with a
+number of benefits:
+
+1. Future-proofing. Since modules are part of the ECMAScript specification, it
+   is clear that they become commonplace fairly soon. Ember has long been
+   pushing the approach, this RFC just continues what's been started a couple
+   years ago with Ember AppKit, followed by Ember CLI.
+2. Makes possible to distribute ember in smaller packages. This could further
+   slim down Ember's infamous byte size - but note that this may not mean
+   completely separate NPM modules. Some libraries are even worthy of
+   distributed by themselves, for example the module-based resolver is already a
+   separate package. Some notable others are also simply external libraries
+   packaged alongside Ember.
+3. Treeshaking. Since dependencies could be determined at compile-time based on
+   the `import` statements, parts of Ember that are not used by an application
+   would not end up in the final build. Think of including only
+   `HistoryLocation` when an app doesn't need to support hashchange-only
+   browsers.
+
+# Detailed design
+
+## End-user syntax
+
+Take the following Ember CLI example:
+
+```js
+import Ember from 'ember';
+
+export Ember.Component.extend({
+  someMethod(arg) {
+    return Ember.isBlank(arg);
+  }
+});
+```
+
+Here, both `Ember.Component` and `Ember.isBlank` could be `import`ed:
+
+```js
+import Component from 'ember-component';
+import { isBlank } from 'ember-utils';
```

Comment:
Could you point to an sample implementation? I'm happy to dig in, since I feel introducing more than one way to access Ember stuff brings only confusion to newcomers especially.